ledger/participant-local-store/src/main/scala/com/daml/platform/localstore/InMemoryUserManagementStore.scala

```diff
@@ -56,59 +58,66 @@ class InMemoryUserManagementStore(createAdmin: Boolean = true) extends UserManag
   override def updateUser(
       userUpdate: UserUpdate
   )(implicit loggingContext: LoggingContext): Future[Result[User]] = {
-    withUser(userUpdate.id) { userInfo =>
-      val updatedPrimaryParty = userUpdate.primaryPartyUpdateO.getOrElse(userInfo.user.primaryParty)
-      val updatedIsDeactivated =
-        userUpdate.isDeactivatedUpdateO.getOrElse(userInfo.user.isDeactivated)
-      val existingAnnotations = userInfo.user.annotations
-      val updatedAnnotations =
-        userUpdate.metadataUpdate.annotationsUpdateO.fold(existingAnnotations) { newAnnotations =>
-          LocalAnnotationsUtils.calculateUpdatedAnnotations(
-            newValue = newAnnotations,
-            existing = existingAnnotations,
-          )
-        }
-      val currentResourceVersion = userInfo.user.resourceVersion
-      val newResourceVersionEither = userUpdate.metadataUpdate.resourceVersionO match {
-        case None => Right(currentResourceVersion + 1)
-        case Some(requestResourceVersion) =>
-          if (requestResourceVersion == currentResourceVersion) {
-            Right(currentResourceVersion + 1)
-          } else {
-            Left(UserManagementStore.ConcurrentUserUpdate(userUpdate.id))
+    withUser(userUpdate.id, Ref.IdentityProviderId.Default) {
```

Comment:
This Default is not always representing the truth. We actually have sometimes the big admin trying to change the identity provider assignment. Sometimes we have IDP-admin trying to change some other field